instrumentation/delayed_job/lib/opentelemetry/instrumentation/delayed_job/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'delayed/plugin'
+
+module OpenTelemetry
+  module Instrumentation
+    module DelayedJob
+      module Middlewares
+        # Delayed Job plugin that instruments invoke_job and other hooks
+        class TracerMiddleware < Delayed::Plugin
+          class << self
+            def instrument_enqueue(job, &block)
+              return block.call(job) unless enabled?
+
+              tracer.in_span('delayed_job.enqueue', kind: :producer) do |span|
+                yield job
+                add_attributes(span, job)
+                add_events(span, job)
+              end
+            end
+
+            def instrument_invoke(job, &block) # rubocop:disable Metrics/AbcSize
+              return block.call(job) unless enabled?
+
+              tracer.in_span('delayed_job.invoke', kind: :consumer) do |span|
+                add_attributes(span, job)
+                span.set_attribute('delayed_job.attempts', job.attempts) if job.attempts
+                span.set_attribute('delayed_job.locked_by', job.locked_by) if job.locked_by
+                add_events(span, job)
+                begin
+                  yield job
+                rescue StandardError => e
+                  span.set_attribute('error', true)
+                  span.set_attribute('error.kind', e.class.name)
+                  span.set_attribute('message', e.message&.[](0...120))
+                  raise e
+                end
```

Comment:
This is already handled in `tracer.in_span`, so you don't need it here.

Reply:
Done